src/rust/engine/process_execution/src/local.rs

```diff
@@ -704,11 +706,15 @@ pub async fn prepare_workdir(
   // non-determinism when paths overlap: see the method doc.
   let store2 = store.clone();
   let workdir_path_2 = workdir_path.clone();
+  let mut mutable_paths = req.output_files.clone();
+  mutable_paths.append(&mut req.output_directories.clone());
```

Comment:
This is the first time I'm seeing `.append(&mut collection)`... I'm more used to `.extend(iterator)`.